lib/hal/debug.c

```diff
@@ -42,6 +42,11 @@ static void drawChar(unsigned char c, int x, int y, int fgColour, int bgColour)
 		mask = 0x01;
 #else
 		mask = 0x80;
+#endif
+
+		// This loop draws up to 1 byte (8 bit) per line; so 8 pixels at most
+#if FONT_WIDTH > 8
+#error Font can't be wider than 8 pixels
 #endif
```

Comment:
You are checking the value of `FONT_WIDTH` _after_ you already used it for calculations. I can see that in this case it doesn't matter much but looks strange to me?

Reply:
I'm only checking it here, because the problems only arise here. The code above is fine.

In the code above we consider 1 byte (= 8 bit) per 8 width-pixels. The old code looked like it handled this, but it didn't. It assumed `FONT_WIDTH` bytes per symbol, which was 8; that luckily happened to match `FONT_HEIGHT` which was also 8. So it correctly skipped 8 bytes per pixel (8 rows x 1 byte per row = 8x8 symbol), but the math was wrong. The new code fixes the math by rounding up to the necessary number of bytes per line using `((FONT_WIDTH+7)/8)`, then using `FONT_HEIGHT` to account for the number of lines per symbol.
So the code above will now handle *any* `FONT_WIDTH` and `FONT_HEIGHT` (until numbers overflow - but let's ignore that for now).
Hence, there's no error there.

However, as indicated by the new comment for this loop: If we have anything other than 8 for this loop, then we'd need to split drawing into groups of 8 bits each (to process each byte per line individually). We'll also have to deal with endianess and more complex mirroring logic then.
Hence, this has the error next to it.

Reply:
I think it's fine to have the check there. I'd just like to mention that the C11 `static_assert` / `_Static_assert` is also an option for checks like this.

Reply:
I'd like to avoid `static_assert` for this. There's no reason to bump compiler requirements for a minor change like this.